### ISSUE TITLE
twisted.web.proxy creates responses with too many headers

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4277]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4277|
|Type|release blocker: regression|
|Created|2010-02-15 19:42:06Z|
|Branch|https://github.com/twisted/twisted/tree/duplicate-headers-4277-2|

Since [#2677](https://github.com/twisted/twisted/issues/2677) was resolved, `twisted.web.proxy` now includes every header it receives in the response it sends back to the client.

Unfortunately this results in some duplication in headers.  Twisted Web adds the _Server_, _Date_, and _Content-Type_ headers unconditionally, so the possibility exists for two of each of these to be included in the response.

<details><summary>Searchable metadata</summary>

```
trac-id__4277 4277
type__release_blocker__regression release blocker: regression
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_duplicate_headers_4277_2 branches/duplicate-headers-4277-2
branch_author__jml jml
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1266262926000000 1266262926000000
changetime__1266787519000000 1266787519000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

(In [28303]) Branching to 'duplicate-headers-4277'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

(In [28309]) Branching to 'duplicate-headers-4277-2'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml removed owner|
|-|-|

Itamar & I have got a fix for this.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @jml|
|-|-|

Great, +1.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml set status to `closed`|
|-|-|

(In [28330]) Merge in duplicate-headers-4277-2

 * Author: jml, itamar
 * Reviewer: therve
 * Fixes [#4277](https://github.com/twisted/twisted/issues/4277)

Proxy no longer duplicates headers.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
